File: src/types/template.ts
import type { KanbanAutomationRule, KanbanBoard, KanbanFilterSpec } from './kanban';

export type TemplateSource = 'builtin' | 'vault' | 'app';

export interface KanbanTemplate {
  kind: 'kanban';
  name: string;
  source: TemplateSource;
  hash: string;
  updatedAt: number;
  board: KanbanBoard;
}

export interface KanbanFilterPreset {
  kind: 'kanban-filter';
  name: string;
  source: TemplateSource;
  updatedAt: number;
  spec: KanbanFilterSpec;
}

export interface KanbanAutomationPreset {
  kind: 'kanban-automation';
  name: string;
  source: TemplateSource;
  updatedAt: number;
  rule: KanbanAutomationRule;
}

File: src-tauri/src/models/template.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
pub enum TemplateSource {
    Builtin,
    Vault,
    App,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct KanbanTemplate {
    pub kind: String,
    pub name: String,
    pub source: TemplateSource,
    pub hash: String,
    pub updated_at: u64,
    pub board: Value,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct KanbanFilterPreset {
    pub kind: String,
    pub name: String,
    pub source: TemplateSource,
    pub updated_at: u64,
    pub spec: Value,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct KanbanAutomationPreset {
    pub kind: String,
    pub name: String,
    pub source: TemplateSource,
    pub updated_at: u64,
    pub rule: Value,
}

#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
pub enum NoteSnippetScope {
    Vault,
    App,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct NoteSnippet {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    pub scope: NoteSnippetScope,
    pub category: Option<String>,
    pub body: String,
    pub updated_at: u64,
}
